static/src/js/components/TemplateList/TemplateList.jsx

```diff
@@ -0,0 +1,145 @@
+import React, {
+  useCallback,
+  useEffect,
+  useState
+} from 'react'
+import PropTypes from 'prop-types'
+import { Link } from 'react-router-dom'
+import { Col, Row } from 'react-bootstrap'
+import pluralize from 'pluralize'
+import Page from '../Page/Page'
+import useAppContext from '../../hooks/useAppContext'
+import getTemplates from '../../utils/getTemplates'
+import Table from '../Table/Table'
+import EllipsisLink from '../EllipsisLink/EllipsisLink'
+import ErrorBanner from '../ErrorBanner/ErrorBanner'
+
+const TemplateList = ({ templateType }) => {
+  const { user } = useAppContext()
+  const { token } = user
+
+  const [templateList, setTemplateList] = useState([])
+  const [errors, setErrors] = useState()
+  const [loading, setLoading] = useState(true)
+
+  const { providerId } = user
+
+  useEffect(() => {
+    const fetchTemplates = async () => {
+      const { response, error } = await getTemplates(providerId, token)
+
+      setErrors(error)
+      setTemplateList(response)
+      setLoading(false)
+    }
+
+    fetchTemplates()
+  }, [])
+
+  const buildEllipsisLinkCell = useCallback((cellData, rowData) => {
```

Comment:
What is the purpose of the `useCallback` with an empty dependency array? My understanding is that this is the equivalent of `useMemo` because without a dependency array the same 'version' of the method is used every single time.

Reply:
From what I understand, when I provide an empty dependency it means that the callback function should never change as long as the component is mounted. Which is what I want, I want the function to remain constant through out the component lifecycle.